pkg/sources/github/github.go

```diff
@@ -132,173 +141,192 @@ func (s *Source) Init(aCtx context.Context, name string, jobId, sourceId int64,
-	}
-
-	var apiClient *github.Client
-	switch cred := s.conn.GetCredential().(type) {
-	case *sourcespb.GitHub_Unauthenticated:
-		apiClient = github.NewClient(s.httpClient)
-		if len(s.orgs) > 30 {
-			log.Warn("You may experience rate limiting when using the unauthenticated GitHub api. Consider using an authenticated scan instead.")
+	for _, org := range s.orgs {
+		errOrg := s.addReposByOrg(ctx, apiClient, org)
+		errUser := s.addReposByUser(ctx, apiClient, org)
+		if errOrg != nil && errUser != nil {
+			log.WithError(errOrg).Error("error fetching repos for org or user: ", org)
 		}
+	}
+	return apiClient
+}
 
-		if len(s.orgs) > 0 {
-			for _, org := range s.orgs {
-				errOrg := s.addReposByOrg(ctx, apiClient, org)
-				errUser := s.addReposByUser(ctx, apiClient, org)
-				if errOrg != nil && errUser != nil {
-					log.WithError(errOrg).Error("error fetching repos for org or user: ", org)
-				}
-			}
-		}
-	case *sourcespb.GitHub_Token:
-		// needed for clones
-		s.token = cred.Token
+func (s *Source) enumerateWithToken(ctx context.Context, apiEndpoint, token string) (*github.Client, error) {
+	// needed for clones
+	s.token = token
 
-		// needed to list repos
-		ts := oauth2.StaticTokenSource(
-			&oauth2.Token{AccessToken: cred.Token},
-		)
-		tc := oauth2.NewClient(context.TODO(), ts)
+	// needed to list repos
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(context.TODO(), ts)
 
-		var err error
-		// If we're using public github, make a regular client.
-		// Otherwise make an enterprise client
-		if apiEndpoint == "https://api.github.com" {
-			apiClient = github.NewClient(tc)
-		} else {
-			apiClient, err = github.NewEnterpriseClient(apiEndpoint, apiEndpoint, tc)
-			if err != nil {
-				return errors.New(err)
-			}
+	var err error
+	// If we're using public github, make a regular client.
+	// Otherwise make an enterprise client
+	var apiClient *github.Client
+	if apiEndpoint == "https://api.github.com" {
+		apiClient = github.NewClient(tc)
+	} else {
+		apiClient, err = github.NewEnterpriseClient(apiEndpoint, apiEndpoint, tc)
+		if err != nil {
+			return nil, errors.New(err)
 		}
+	}
 
-		// TODO: this should support scanning users too
+	// TODO: this should support scanning users too
 
-		specificScope := false
+	specificScope := false
 
-		if len(s.repos) > 0 {
-			specificScope = true
-		}
+	if len(s.repos) > 0 {
+		specificScope = true
+	}
 
-		if len(s.orgs) > 0 {
-			specificScope = true
-			for _, org := range s.orgs {
-				errOrg := s.addReposByOrg(ctx, apiClient, org)
-				errUser := s.addReposByUser(ctx, apiClient, org)
-				if errOrg != nil && errUser != nil {
-					log.WithError(errOrg).Error("error fetching repos for org or user: ", org)
-				}
+	if len(s.orgs) > 0 {
+		specificScope = true
+		for _, org := range s.orgs {
+			errOrg := s.addReposByOrg(ctx, apiClient, org)
+			errUser := s.addReposByUser(ctx, apiClient, org)
+			if errOrg != nil || errUser != nil {
```

Comment:
As mentioned in the overview, this is the bug-fix (changed `&&` to `||` in this if-statement).

Reply:
Please correct me if I'm mistaken in making this change.

Reply:
It's a bit unfortunate on how this was setup, but one of these will always error. Both are attempted  because users and organizations have the same url format and can be put in the same field.